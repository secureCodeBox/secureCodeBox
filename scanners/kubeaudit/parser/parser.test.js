// SPDX-FileCopyrightText: the secureCodeBox authors
//
// SPDX-License-Identifier: Apache-2.0

const fs = require("fs");
const util = require("util");
const {
  validateParser,
} = require("@securecodebox/parser-sdk-nodejs/parser-utils");

// eslint-disable-next-line security/detect-non-literal-fs-filename
const readFile = util.promisify(fs.readFile);

const { parse } = require("./parser");

test("should properly parse kubeaudit juice-shop results", async () => {
  const fileContent = await readFile(
    __dirname + "/__testFiles__/juice-shop.jsonl",
    {
      encoding: "utf8",
    }
  );
  const findings = await parse(fileContent);
  await expect(validateParser(findings)).resolves.toBeUndefined();

  expect(findings).toMatchSnapshot();
});

test("should properly parse empty kubeaudit jsonl file", async () => {
  const jsonContent = await readFile(
    __dirname + "/__testFiles__/test-empty-report.jsonl",
    {
      encoding: "utf8",
    }
  );
  const findings = await parse(jsonContent);
  await expect(validateParser(findings)).resolves.toBeUndefined();
  expect(findings).toMatchInlineSnapshot(`[]`);
});
