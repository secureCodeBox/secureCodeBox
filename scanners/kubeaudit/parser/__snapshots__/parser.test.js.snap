// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should properly parse kubeaudit juice-shop results 1`] = `
[
  {
    "attributes": {},
    "category": "Automounted ServiceAccount Token",
    "description": "Default service account with token mounted. automountServiceAccountToken should be set to 'false' on either the ServiceAccount or on the PodSpec or a non-default service account should be used.",
    "location": null,
    "name": "Default ServiceAccount uses Automounted Service Account Token",
    "osi_layer": "NOT_APPLICABLE",
    "severity": "LOW",
  },
  {
    "attributes": {
      "capability": "AUDIT_WRITE",
      "container": "juice-shop",
    },
    "category": "Capability Not Dropped",
    "description": "Capability not dropped. Ideally, the capability drop list should include the single capability 'ALL' which drops all capabilities.",
    "location": "container://juice-shop",
    "name": "Capability 'AUDIT_WRITE' Not Dropped",
    "osi_layer": "NOT_APPLICABLE",
    "severity": "LOW",
  },
  {
    "attributes": {
      "capability": "CHOWN",
      "container": "juice-shop",
    },
    "category": "Capability Not Dropped",
    "description": "Capability not dropped. Ideally, the capability drop list should include the single capability 'ALL' which drops all capabilities.",
    "location": "container://juice-shop",
    "name": "Capability 'CHOWN' Not Dropped",
    "osi_layer": "NOT_APPLICABLE",
    "severity": "LOW",
  },
  {
    "attributes": {
      "capability": "DAC_OVERRIDE",
      "container": "juice-shop",
    },
    "category": "Capability Not Dropped",
    "description": "Capability not dropped. Ideally, the capability drop list should include the single capability 'ALL' which drops all capabilities.",
    "location": "container://juice-shop",
    "name": "Capability 'DAC_OVERRIDE' Not Dropped",
    "osi_layer": "NOT_APPLICABLE",
    "severity": "LOW",
  },
  {
    "attributes": {
      "capability": "FOWNER",
      "container": "juice-shop",
    },
    "category": "Capability Not Dropped",
    "description": "Capability not dropped. Ideally, the capability drop list should include the single capability 'ALL' which drops all capabilities.",
    "location": "container://juice-shop",
    "name": "Capability 'FOWNER' Not Dropped",
    "osi_layer": "NOT_APPLICABLE",
    "severity": "LOW",
  },
  {
    "attributes": {
      "capability": "FSETID",
      "container": "juice-shop",
    },
    "category": "Capability Not Dropped",
    "description": "Capability not dropped. Ideally, the capability drop list should include the single capability 'ALL' which drops all capabilities.",
    "location": "container://juice-shop",
    "name": "Capability 'FSETID' Not Dropped",
    "osi_layer": "NOT_APPLICABLE",
    "severity": "LOW",
  },
  {
    "attributes": {
      "capability": "KILL",
      "container": "juice-shop",
    },
    "category": "Capability Not Dropped",
    "description": "Capability not dropped. Ideally, the capability drop list should include the single capability 'ALL' which drops all capabilities.",
    "location": "container://juice-shop",
    "name": "Capability 'KILL' Not Dropped",
    "osi_layer": "NOT_APPLICABLE",
    "severity": "LOW",
  },
  {
    "attributes": {
      "capability": "MKNOD",
      "container": "juice-shop",
    },
    "category": "Capability Not Dropped",
    "description": "Capability not dropped. Ideally, the capability drop list should include the single capability 'ALL' which drops all capabilities.",
    "location": "container://juice-shop",
    "name": "Capability 'MKNOD' Not Dropped",
    "osi_layer": "NOT_APPLICABLE",
    "severity": "LOW",
  },
  {
    "attributes": {
      "capability": "NET_BIND_SERVICE",
      "container": "juice-shop",
    },
    "category": "Capability Not Dropped",
    "description": "Capability not dropped. Ideally, the capability drop list should include the single capability 'ALL' which drops all capabilities.",
    "location": "container://juice-shop",
    "name": "Capability 'NET_BIND_SERVICE' Not Dropped",
    "osi_layer": "NOT_APPLICABLE",
    "severity": "LOW",
  },
  {
    "attributes": {
      "capability": "NET_RAW",
      "container": "juice-shop",
    },
    "category": "Capability Not Dropped",
    "description": "Capability not dropped. Ideally, the capability drop list should include the single capability 'ALL' which drops all capabilities.",
    "location": "container://juice-shop",
    "name": "Capability 'NET_RAW' Not Dropped",
    "osi_layer": "NOT_APPLICABLE",
    "severity": "LOW",
  },
  {
    "attributes": {
      "capability": "SETFCAP",
      "container": "juice-shop",
    },
    "category": "Capability Not Dropped",
    "description": "Capability not dropped. Ideally, the capability drop list should include the single capability 'ALL' which drops all capabilities.",
    "location": "container://juice-shop",
    "name": "Capability 'SETFCAP' Not Dropped",
    "osi_layer": "NOT_APPLICABLE",
    "severity": "LOW",
  },
  {
    "attributes": {
      "capability": "SETGID",
      "container": "juice-shop",
    },
    "category": "Capability Not Dropped",
    "description": "Capability not dropped. Ideally, the capability drop list should include the single capability 'ALL' which drops all capabilities.",
    "location": "container://juice-shop",
    "name": "Capability 'SETGID' Not Dropped",
    "osi_layer": "NOT_APPLICABLE",
    "severity": "LOW",
  },
  {
    "attributes": {
      "capability": "SETPCAP",
      "container": "juice-shop",
    },
    "category": "Capability Not Dropped",
    "description": "Capability not dropped. Ideally, the capability drop list should include the single capability 'ALL' which drops all capabilities.",
    "location": "container://juice-shop",
    "name": "Capability 'SETPCAP' Not Dropped",
    "osi_layer": "NOT_APPLICABLE",
    "severity": "LOW",
  },
  {
    "attributes": {
      "capability": "SETUID",
      "container": "juice-shop",
    },
    "category": "Capability Not Dropped",
    "description": "Capability not dropped. Ideally, the capability drop list should include the single capability 'ALL' which drops all capabilities.",
    "location": "container://juice-shop",
    "name": "Capability 'SETUID' Not Dropped",
    "osi_layer": "NOT_APPLICABLE",
    "severity": "LOW",
  },
  {
    "attributes": {
      "capability": "SYS_CHROOT",
      "container": "juice-shop",
    },
    "category": "Capability Not Dropped",
    "description": "Capability not dropped. Ideally, the capability drop list should include the single capability 'ALL' which drops all capabilities.",
    "location": "container://juice-shop",
    "name": "Capability 'SYS_CHROOT' Not Dropped",
    "osi_layer": "NOT_APPLICABLE",
    "severity": "LOW",
  },
  {
    "attributes": {
      "container": "juice-shop",
    },
    "category": "Non Root User Not Enforced",
    "description": "runAsNonRoot is not set in container SecurityContext nor the PodSecurityContext. It should be set to 'true' in at least one of the two.",
    "location": "container://juice-shop",
    "name": "NonRoot User not enforced for Container",
    "osi_layer": "NOT_APPLICABLE",
    "severity": "MEDIUM",
  },
  {
    "attributes": {
      "container": "juice-shop",
    },
    "category": "Non ReadOnly Root Filesystem",
    "description": "readOnlyRootFilesystem is not set in container SecurityContext. It should be set to 'true'.",
    "location": "container://juice-shop",
    "name": "Container Uses a non ReadOnly Root Filesystem",
    "osi_layer": "NOT_APPLICABLE",
    "severity": "LOW",
  },
  {
    "attributes": {
      "Namespace": "default",
    },
    "category": "No Default Deny NetworkPolicy",
    "description": "Namespace is missing a default deny ingress and egress NetworkPolicy.",
    "location": "namespace://default",
    "name": "Namespace "default" is missing a Default Deny NetworkPolicy",
    "osi_layer": "NOT_APPLICABLE",
    "severity": "MEDIUM",
  },
]
`;
